### PR TITLE
node: add dump-rpcs command

### LINKED ISSUE #1810: Guardiand should have a simple way to list RPC nodes
Currently the only way to determine RPC nodes is parsing through logs or looking at the startup command/script/unitfile.

Parsing through logs is unwieldy.
If the startup command/script/unitfile has changed on disk but the service has not been restarted then it's no longer a reliable source of which RPC nodes are active.

having something along the lines of "guardiand admin list-rpc" would be a definitive source of which RPC nodes are being referred to.

### PR DESCRIPTION
**Pull request by panoel**
Fixes #1810

This is an example of the output:
``` > kubectl exec guardian-0 -- /guardiand admin dump-rpcs --socket /tmp/admin.sock
acalaRPC = [ws://eth-devnet:8545]
algorandIndexerRPC = []
algorandAlgodRPC = []
aptosRPC = []
arbitrumRPC = [ws://eth-devnet:8545]
auroraRPC = [ws://eth-devnet:8545]
avalancheRPC = [ws://eth-devnet:8545]
bscRPC = [ws://eth-devnet:8545]
celoRPC = [ws://eth-devnet:8545]
ethRPC = [ws://eth-devnet:8545]
fantomRPC = [ws://eth-devnet:8545]
karuraRPC = [ws://eth-devnet:8545]
klaytnRPC = [ws://eth-devnet:8545]
moonbeamRPC = [ws://eth-devnet:8545]
nearRPC = []
neonRPC = [ws://eth-devnet:8545]
oasisRPC = [ws://eth-devnet:8545]
polygonRPC = [ws://eth-devnet:8545]
pythnetRPC = []
solanaRPC = []
terraWS = []
terraLCD = []
terra2WS = []
terra2LCD = []
wormchainWS = []
wormchainLCD = []
xplaWS = []
xplaLCD = []

### CODE REVIEW

**Review by bruce-riley — APPROVED** [review]

File: node/cmd/guardiand/adminserver.go [hunk 1]
```diff
@@ -689,3 +689,39 @@ func (s *nodePrivilegedService) SignExistingVAA(ctx context.Context, req *nodev1
 
 	return &nodev1.SignExistingVAAResponse{Vaa: newVAABytes}, nil
 }
+
+func (s *nodePrivilegedService) DumpRPCs(ctx context.Context, req *nodev1.DumpRPCsRequest) (*nodev1.DumpRPCsResponse, error) {
```
[the remark this answers is not in the document]
Comment (panoel) [reply node/cmd/guardiand/adminserver.go:693]: Fixed

File: node/cmd/guardiand/adminclient.go [hunk 2]
```diff
@@ -375,6 +384,26 @@ func runSendObservationRequest(cmd *cobra.Command, args []string) {
...
+		log.Fatalf("failed to get admin client: %v", err)
+	}
+	defer conn.Close()
+
+	resp, err := c.DumpRPCs(ctx, &nodev1.DumpRPCsRequest{})
+	if err != nil {
+		log.Fatalf("failed to run dump-rpcs: %s", err)
+	}
+
+	for node, rpc := range resp.Response {
```
Comment (bruce-riley) [inline node/cmd/guardiand/adminclient.go:402]: Extreme nit: The key is not really a node, it's a parameter (and in some cases there are more than one parameter per node).

**Review by bruce-riley — APPROVED** [review]

File: node/cmd/guardiand/adminclient.go (hunk 2 above)
Comment (panoel) [reply node/cmd/guardiand/adminclient.go]: Fixed

**Review by bruce-riley — APPROVED** [review]

**Review by hendrikhofstadt — APPROVED** [review]

File: node/cmd/guardiand/adminserver.go [hunk 3]
```diff
@@ -689,3 +689,40 @@ func (s *nodePrivilegedService) SignExistingVAA(ctx context.Context, req *nodev1
...
+	rpcMap["ethRPC"] = *ethRPC
+	rpcMap["fantomRPC"] = *fantomRPC
+	rpcMap["karuraRPC"] = *karuraRPC
+	rpcMap["klaytnRPC"] = *klaytnRPC
+	rpcMap["moonbeamRPC"] = *moonbeamRPC
+	rpcMap["nearRPC"] = *nearRPC
+	rpcMap["neonRPC"] = *neonRPC
+	rpcMap["oasisRPC"] = *oasisRPC
+	rpcMap["polygonRPC"] = *polygonRPC
+	rpcMap["pythnetRPC"] = *pythnetRPC
```
Comment (evan-gray) [inline node/cmd/guardiand/adminserver.go:714]: 🤔 can handle in a follow up, but isn't there a `pythnetWS` now? wonder if there's a cleaner way to keep these in sync with the flags.

**Review by bruce-riley — APPROVED** [review]

**Review by hendrikhofstadt — APPROVED** [review]

**Pull request merged**